proposals/safe-fixed-size-buffers.md

```diff
@@ -0,0 +1,399 @@
+
+#### Fixed-size buffer element access
+
+For a fixed-size buffer element access, the *primary_no_array_creation_expression* of the *element_access* must be a variable or value of a fixed-size buffer type. Furthermore, the *argument_list* of a fixed-size buffer element access is not allowed to contain named arguments. The *argument_list* must contain a single expression, and the expression must be 
+- either of type `int`, or
+- must be implicitly convertible to `int`, or
+- must be of type ```System.Index```, or 
+- must be of type ```System.Range```. 
+
+If *primary_no_array_creation_expression* is a variable, result of evaluating a fixed-size buffer element access is result of indexing into a ```System.Span<T>``` returned by *AsSpan* method with the same argument. 
+
+If *primary_no_array_creation_expression* is a value, result of evaluating a fixed-size buffer element access is result of indexing into a ```System.ReadOnlySpan<T>``` returned by *AsReadOnlySpan* with the same argument. 
+
+Indexing into a fixed-size buffer with a constant expression outside of the declared fixed-size buffer bounds is a compile time error.
+
+The *safe-to-escape* scope of the value will be equal to the *safe-to-escape* scope of the *primary_no_array_creation_expression*,
+just as it would if the backing data was accessed as a field.
+
+### Conversions
+
+A new conversion from expression will be added.
+
+There is an implicit conversion from expression representing a variable of a fixed-size buffer type to the following types:
+- ```System.Span<T>``` - the result of conversion is a span returned by *AsSpan* method.
+- ```System.ReadOnlySpan<T>``` - the result of conversion is a span returned by *AsReadOnlySpan* method.
+
+There is an implicit conversion from expression representing a value of a fixed-size buffer type to ```System.ReadOnlySpan<T>``` type.
+The result of conversion is a span returned by *AsReadOnlySpan* method.
+
+### List patterns
+
+[List patterns](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-11.0/list-patterns.md) will not be supported for instances
+of fixed-size buffer types. In order to be able to take advantage of list patterns on fixed-size buffer types, a target expression must be
+explicitly converted either to ```System.Span<T>```, or to ```System.ReadOnlySpan<T>``` type first. The reason is the fact that pattern evaluation 
+liberally makes copies of the target expression, and it would be impossble to slice the copy into ```System.Span<T>```, but intent of the user might
```

Comment:
```suggestion
liberally makes copies of the target expression, and it would be impossible to slice the copy into ```System.Span<T>```, but intent of the user might
```